src/MSMToolsCluster.jl

```diff
@@ -0,0 +1,259 @@
+    if m < n_closed_components
+        throw(error("Number of metastable states m=$m is too small. Transition matrix " *
+              "has $n_closed_components disconnected components."))
+    end
+
+    closed_components_Psub = []
+    closed_components_ev = []
+    closed_components_enum = []
+    for i in 1:n_closed_components
+        component = closed_components[i]
+        Psub = P[component, component] # compute eigenvalues in submatrix
+        push!(closed_components_Psub, Psub)
+        push!(closed_components_ev, eigen(Psub).values)
+        push!(closed_components_enum, fill(i, length(component)))
+    end
+
+    closed_components_ev_flat = vcat(closed_components_ev...)
+    closed_components_enum_flat = vcat(closed_components_enum...)
+
+    # cluster each component
+    component_indexes = closed_components_enum_flat[sortperm(closed_components_ev_flat)][1:m]
+    ipcca = 1
+    for i in 1:n_closed_components
+        component = closed_components[i]
+        m_by_component = count(x -> x == i, component_indexes) # number of PCCA states per component
+
+        if m_by_component == 1 # trivial case
+            chi[component, ipcca] .= 1.0
+            ipcca += 1
+        elseif m_by_component > 1
+            chi[component, ipcca:(ipcca + m_by_component - 1)] .= pcca_connected(
+                closed_components_Psub[i], m_by_component, π == nothing ? nothing : π[component]
+            )
+            ipcca += m_by_component
+        else
+            throw(error("Did not expect component $i to have $m_by_component."))
+        end
+    end
+
+    # assign transition states
+    if !isempty(transition_states)
+        # make all closed states absorbing, so we can see which closed state we hit first
+        Pabs = copy(P)
+        Pabs[closed_states, :] .= 0.0
+        Pabs[closed_states, closed_states] .= 1.0
+        for i in closed_states
+            h = hitting_probability(Pabs, i) # hitting probability to each closed state
+            for j in transition_states
+                chi[j, :] .+= h[j] .* chi[i, :]
+            end
+        end
+    end
+
+    nmeta = count(x -> x != 0, sum(chi, dims=1))
+    if nmeta < m
+        throw(error("Requested $m metastable states, but transition matrix only has $nmeta. " *
+                "Try requesting fewer metastable states."))
+    end
+
+    return chi
+end
+
+# Reimplementation of PCCA+ spectral clustering method with optimized memberships (assumes fully connected transition matrix)
+# https://github.com/deeptime-ml/deeptime/blob/a6ac0b93a55d688fe8f3af119680105763366220/deeptime/markov/tools/analysis/dense/_pcca.py#L197
+function pcca_connected(P, n, pi=nothing)
+    labels = connected_sets(P)
+    n_components = length(labels)
+    if n_components > 1
+        throw(error("Unable to use pcca_connected, transition matrix is not fully connected."))
+    end
+
+    if pi === nothing
+        pi = stationary_distribution(P)
```

Comment:
Instead of setting the default value of `pi` as `nothing` in the function signature it migth be best to use the default `stationary_distribution(P)` like
```
 function pcca_connected(P, n, pi=stationary_distribution(P)
```

Reply:
fixed in 90de4adbea72feee7b41b1aa2a1b617ca178219e